controllers/reconcilers/configuration/prometheus.go

```diff
@@ -150,7 +170,10 @@ func (r *Reconciler) getOpenshiftMonitoringCredentials(ctx context.Context) (str
 		return "", "", err
 	}
 
-	return ds.Sources[0].BasicAuthUser, ds.Sources[0].BasicAuthPassword, nil
+	if !isDatasourceV2 {
+		return ds.Sources[0].BasicAuthUser, ds.Sources[0].BasicAuthPassword, nil
```

Comment:
Is there any chance that the array's could have greater or less than 1 item in it?

Reply:
I think the logic is to just pick the first item in the array. Not sure of use cases where array would contain more than one item. Would you have an insight here @pb82

Reply:
We should check that the array is not empty because in that case this results in a nil pointer exception. If the array is empty we should log an error and continue without federation.